Implementation of Physics Bodies - PART 1

aaronlamchops opened the pull request:
# Implementations:

- Added a command on the 'G' Key that allows you to turn off the graphics allowing us to debug to physics bodies.
- Added a command on the 'H' Key that turns the graphics back on.
- Added physics bodies to all the walls and to the character.
- Implemented a different style for the camera system that uses Context.Translate(x,y). Uses less lines and works with the physics bodies better.
- Implemented more functions for the physics system and added more character functions to access properties that use physics.
- Fixed all merge conflicts and centered the dust particles to character.body.position.x and y

## NOTE:
_I add a different keyboard listener off of our input.js that uses the WASD, but i implemented all the existing movement to what we have currently, which uses the arrow keys. So all in all, we have two methods of receiving input via the arrow keys and WASD. I think we should switch to WASD because it might be more natural when we implement attacking, but for right now the arrow keys has all the right properties tied to it._

Commit:
Updating code

Commit:
Fixed merge conflicts and implemented bodies on character and walls

bumbledog reviewed (changes requested):
Overall, solid commit. Before we merge it though, be careful to set the firstRender and stuff to be true in the initialize instead of the game object, or when we start a new level it won't refresh correctly.
Also, is there a reason we need a separate object representing physics walls and actual walls? Seems repetitive if the objects will end up being identical.

bumbledog commented:
Also, the character bounces off the wall before it actually gets close. Could you shrink the body around the character?

Commit:
Made changes to firstRender

aaronlamchops commented:
Okay, I changed the way firstRender is set to true by switching it to being called true in the Initialize of the gameloop. 

As for the representation of the physicsWalls, its an object that holds the bodies. I implemented that object so that i wouldn't through off the algorithm for placing edges in the maze. If we delete the edges object, then we would have to change the way that the algorithm factors for placing edges. Instead of checking NULL we would have to check for a physics body instead. Personally, i think that is fine with another object in the maze to hold the actual bodies of the walls.

As for the character bouncing off the wall before it hits the edge, that is because the edges are rendered at a very small thickness. The character is actually hitting the body of the walls and not the thin edge. We will eventually want the walls that thick so that we can render a different sprite over the walls.
